orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/render/tags/PipelineIdTag.java

```diff
@@ -0,0 +1,85 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipelinetemplate.v1schema.render.tags;
+import com.hubspot.jinjava.interpret.*;
+import com.hubspot.jinjava.lib.tag.Tag;
+import com.hubspot.jinjava.tree.TagNode;
+import com.hubspot.jinjava.util.HelperStringTokenizer;
+import com.netflix.spinnaker.orca.front50.Front50Service;
+import com.netflix.spinnaker.orca.pipelinetemplate.exceptions.TemplateRenderException;
+import com.netflix.spinnaker.orca.pipelinetemplate.v1schema.model.PipelineId;
+import com.netflix.spinnaker.orca.pipelinetemplate.v1schema.model.PipelineTemplate;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class PipelineIdTag implements Tag {
+  private final Front50Service front50Service;
+
+  public PipelineIdTag(Front50Service front50Service) {
+    this.front50Service = front50Service;
+  }
+
+  @Override
+  public String interpret(TagNode tagNode, JinjavaInterpreter interpreter) {
+    List<String> helper = new HelperStringTokenizer(tagNode.getHelpers()).splitComma(true).allTokens();
+    if (helper.isEmpty()) {
+      throw new TemplateSyntaxException(tagNode.getMaster().getImage(), "Tag 'pipelineId' expects at least a pipeline name: " + helper, tagNode.getLineNumber());
+    }
+
+    Map<String, String> paramPairs = new HashMap<>();
+    helper.forEach(p -> {
+      String[] parts = p.split("=");
+      if (parts.length != 2) {
+        throw new TemplateSyntaxException(tagNode.getMaster().getImage(), "Tag 'pipelineId' expects parameters to be in a 'key=value' format: " + helper, tagNode.getLineNumber());
+      }
+
+      paramPairs.put(parts[0], parts[1]);
+    });
+
+    Context context = interpreter.getContext();
+    String application = paramPairs.getOrDefault("application", (String) context.get("application"));
+
+    PipelineTemplate template = (PipelineTemplate) context.get("pipelineTemplate");
+    if (template == null) {
+      throw new TemplateStateException("Pipeline template missing from jinja context");
+    }
+
+    PipelineId pipeline = template.getPipelineId();
+    List<Map<String, Object>> pipelines = front50Service.getPipelines(pipeline.getApplication());
+    Map<String, Object> matchedPipeline = pipelines
+      .stream()
+      .filter(p -> p.get("name").equals(pipeline.getName()))
+      .findFirst()
+      .orElseThrow(
+        () -> new TemplateRenderException(String.format("Failed to resolve Tag 'pipelineId' with name %s and application %s", pipeline.getName(), application))
+      );
+
+    return (String) matchedPipeline.get("id");
+  }
+
+  @Override
+  public String getEndTagName() {
+    return null;
+  }
+
+  @Override
+  public String getName() {
+    return "pipelineid";
```

Comment:
Jinja wants it lowercase otherwise the tag never registers
```
  /**
   * @return Get name of end tag lowerCase Null if it's a single tag without content.
   */
  String getEndTagName();
```